apps/web/src/app/(site)/costs/page.tsx

```diff
@@ -20,19 +20,50 @@ import EnergyCostsYesterday from "@/components/costs/energy-costs-yesterday-card
 import { Skeleton } from "@energyleaf/ui/skeleton";
 import React, { Suspense } from "react";
 
+import { getSession } from "@/lib/auth/auth.server";
+import { getElectricitySensorIdForUser, getEnergyDataForSensor } from "@/query/energy";
+import { getUserDataHistory } from "@/query/user";
+import { redirect } from "next/navigation";
+
 export const metadata = {
     title: "Kosten | Energyleaf",
 };
 
-export default function CostsPage() {
+export default async function CostsPage() {
+    const { session, user } = await getSession();
+
+    if (!session) {
+        redirect("/");
+    }
+
+    const userId = user.id;
+    const sensorId = await getElectricitySensorIdForUser(userId);
+
+    if (!sensorId) {
+        throw new Error("Sensor ID not found for the user.");
+    }
+
+    const now = new Date();
+    let year = now.getFullYear();
+    let month = now.getMonth() - 2;
+    if (month < 0) {
+        month += 12;
+        year -= 1;
+    }
+    const startDate = new Date(year, month, 1);
+    const endDate = now;
+
+    const energyDataRaw = await getEnergyDataForSensor(startDate, endDate, sensorId);
```

Comment:
Das Laden der Daten ist hier ganz anders als bei Hanno (https://github.com/pgenergy/energyleaf/pull/134). Er macht das in den Card-Components.
Da müsstet ihr euch einmal abstimmen, wie ihr das machen wollt

Reply:
Jo haben wir gerade schon. Ist ja eigentlich besser das zu Beginn einmal zu die Daten zu laden und dann an die Cards zu übergeben anstatt das in jeder Card erneut zu machen